src/client/Client.cc

```diff
@@ -14728,9 +15407,13 @@ int Client::_mknod(Inode *dir, const char *name, mode_t mode, dev_t rdev,
   req->set_inode_owner_uid_gid(perms.uid(), perms.gid());
 
   filepath path;
-  dir->make_nosnap_relative_path(path);
-  path.push_dentry(name);
-  req->set_filepath(path); 
+  Dentry *de;
+  int r = _prepare_req_path(dir, req, path, name, true, &de);
+  if (r < 0) {
+    delete req;
+    return r;
+  }
```

Comment:
ditto

Reply:
So the file name is not encrypted? I'm a little puzzled by why.

Reply:
At first glance, that's how I interpret it. The wording is not clear on name and/or contents. Maybe it's only referring to contents (which is N/A in this case)?

Reply:
This is done.